pkg/virt-controller/watch/migration/migration_test.go

```diff
@@ -2264,25 +2266,128 @@ var _ = Describe("Migration watcher", func() {
+					vmi := newVirtualMachine(fmt.Sprintf("testvmi%d", i), virtv1.Running)
+					migration := newMigration(fmt.Sprintf("testmigration%d", i), vmi.Name, virtv1.MigrationRunning)
+					migration.Spec.Priority = pointer.P(virtv1.PrioritySystemMaintenance)
+					pod := newTargetPodForVirtualMachine(vmi, migration, k8sv1.PodRunning)
+					addMigration(migration)
+					addVirtualMachineInstance(vmi)
+					addPod(pod)
+				}
+
+				By("Executing the controller and expecting the pending migration to have the defined priority")
+				controller.Execute()
+				for i := 0; i < 5; i++ {
+					item, priority, shutdown := controller.Queue.GetWithPriority()
+					Expect(item).To(Equal(fmt.Sprintf("default/testmigration%d", i)))
+					Expect(priority).To(Equal(0))
+					Expect(shutdown).To(BeFalse())
+				}
+				item, priority, shutdown := controller.Queue.GetWithPriority()
+				Expect(item).To(Equal("default/testmigrationpending"))
+				Expect(priority).To(Equal(migrationsutil.PrioritySystemCritical))
+				Expect(shutdown).To(BeFalse())
+			})
+
+			It("should get items in order based on priority", func() {
+				By("Creating 5 running migrations")
+				for i := 0; i < 5; i++ {
+					vmi := newVirtualMachine(fmt.Sprintf("testvmi%d", i), virtv1.Running)
+					migration := newMigration(fmt.Sprintf("testmigration%d", i), vmi.Name, virtv1.MigrationRunning)
+					migration.Spec.Priority = pointer.P(virtv1.PrioritySystemCritical)
+					controller.enqueueMigration(migration)
+				}
+
+				By("Creating 5 critical, 5 maintenance, 5 user triggered and 5 non-defined priority migrations")
+				for i := 0; i < 5; i++ {
+					vmi := newVirtualMachine(fmt.Sprintf("testvmi-user-%d", i), virtv1.Running)
+					migration := newMigration(fmt.Sprintf("test-user-migration-%d", i), vmi.Name, virtv1.MigrationPending)
+					migration.Spec.Priority = pointer.P(virtv1.PriorityUserTriggered)
+					controller.enqueueMigration(migration)
+					vmi = newVirtualMachine(fmt.Sprintf("testvmi-crit-%d", i), virtv1.Running)
+					migration = newMigration(fmt.Sprintf("test-crit-migration-%d", i), vmi.Name, virtv1.MigrationPending)
+					migration.Spec.Priority = pointer.P(virtv1.PrioritySystemCritical)
+					controller.enqueueMigration(migration)
+					vmi = newVirtualMachine(fmt.Sprintf("testvmi-noprio-%d", i), virtv1.Running)
+					migration = newMigration(fmt.Sprintf("test-noprio-migration-%d", i), vmi.Name, virtv1.MigrationPending)
+					controller.enqueueMigration(migration)
+					vmi = newVirtualMachine(fmt.Sprintf("testvmi-maint-%d", i), virtv1.Running)
+					migration = newMigration(fmt.Sprintf("test-maint-migration-%d", i), vmi.Name, virtv1.MigrationPending)
+					migration.Spec.Priority = pointer.P(virtv1.PrioritySystemMaintenance)
+					controller.enqueueMigration(migration)
+				}
+
+				for i := 0; i < 5; i++ {
+					item, priority, shutdown := controller.Queue.GetWithPriority()
+					Expect(item).To(BeEquivalentTo(fmt.Sprintf("default/testmigration%d", i)))
```

Comment:
https://prow.ci.kubevirt.io/view/gs/kubevirt-prow/pr-logs/pull/kubevirt_kubevirt/15887/pull-kubevirt-unit-test/1981305405292154880

```
{ Failure pkg/virt-controller/watch/migration/migration_test.go:2331
Expected
    <string>: default/test-noprio-migration-4
to be equivalent to
    <string>: default/testmigration1
pkg/virt-controller/watch/migration/migration_test.go:2361}
```

I saw your retest after this failure, did you push something that fixed this or is this a race that's still present?

Reply:
I actually tried several times and no race. Let me try again and again

Reply:
I have a very low-level idea of why it happens. But the main issue was the non-use of `ConsistOf`.

Reply:
https://github.com/kubernetes-sigs/controller-runtime/issues/3363